utopia-vscode-extension/src/browserfs-utils.ts

```diff
@@ -0,0 +1,295 @@
+import * as BrowserFS from 'browserfs'
+import { ApiError } from 'browserfs/dist/node/core/api_error'
+import { BFSCallback, BFSOneArgCallback } from 'browserfs/dist/node/core/file_system'
+import { FSModule } from 'browserfs/dist/node/core/FS'
+import Stats from 'browserfs/dist/node/core/node_fs_stats'
+import { uint8Array2Buffer } from 'browserfs/dist/node/core/util'
+import { FileSystemError } from 'vscode'
+import { appendToPath, toUtopiaPath } from './path-utils'
+
+let fs: FSModule
+
+interface WatchConfig {
+  recursive: boolean
+  onCreated: (path: string) => void
+  onModified: (path: string) => void
+  onDeleted: (path: string) => void
+}
+
+let watchTimeout: number | null = null
+let watchedPaths: Map<string, WatchConfig> = new Map()
+let lastModifiedTSs: Map<string, number> = new Map()
+const POLLING_TIMEOUT = 100
+
+// Immediately attempt to initialise BrowserFS
+new Promise<void>((resolve, reject) => {
```

Comment:
why is this wrapped in a Promise? I thought that unhandled Promise rejections cause a big warning saying they are not supported